### DISCUSSION TITLE
php-lxd

### DISCUSSION THREAD

**Discussion by kamzar1** [turtle0x1/LxdMosaic#357] [General]
Due to my opinion, php-lxd is not really necessary to be another dependency here.
All LXD Api needs is a GET,PUT,PATCH,POST over secure curl.
So, you need to read the key, cert, pass the remote server address, request type, api endpoint, which can be defined in one single and very short function to create the curl request and fetch the reply.
Everything else you just pass to that function, if given: data, header.
It has saved me running the composer, GuzzleAdapter, and loading for every action this and that php library, php class ...

**Comment by turtle0x1** [comment]
For someone storing raw `exports` (GB or TB long term)  you are worrying alot about micro optimizations of storage (`kb` / `mb`)

 `php-lxd` is doing exactly what you say, except its broken down for you into classes so you can make calls like;

```
$client-images->all();
$client->containers->all();
$client->projects->all();
``` 
If you prefer something like;
```
function curl(...)
{
   // Handle this
   // Handle that
   /// Oh another thing
}

curl("/1.0/instances", "GET, [], "another_param", "do_the_params_ever_stop", "growing_longer_and_longer");
```

Thats fine, but you dont need to tell me. :smile:

**Comment by turtle0x1** [comment]
> saved me running composer

Modern PHP development basicly demands composer.

While you can get away with doing it all "old school" your just building legacy. 

I understand why people like "old school" this program uses its own "home grown" HTTP router. But "its wrong" by modern standards and a waste of my time to maintain it, I could be using symfony for routing and have less headaches !(although mine has been "stable enough")

**Reply by kamzar1** [reply]
You absolutely right and caught me off guard !
It would be a waste, inventing the wheel again and again.
To be honest, I use frameworks as well, Codeigniter and Laravel could easily integrate third party php libs.
Indeed I deal with clients' storage of nearly half petabyte, storage optimization and Peta-Phobie got me!

**Reply by turtle0x1** [reply]
Your definitely right if you have to install LXDMosaic / php-lxd a few thousand times in a server setting, but im not sure you would need to now!

I would be interested in connecting to find out why LXDMosaic didn't quite fit your use case and if there was anything it could do differently that other people might find interesting, so in the future other people might adopt LXDMosaic :smile:
